runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillStateReader.java

```diff
@@ -472,7 +485,13 @@ public void startBatchAndBlock() {
       }
 
       Windmill.KeyedGetDataRequest request = createRequest(toFetch);
-      Windmill.KeyedGetDataResponse response = server.getStateData(computation, request);
+      Windmill.KeyedGetDataResponse response;
+      try (AutoCloseable readWrapper =
+          readWrapperSupplier == null ? null : readWrapperSupplier.get()) {
+        response = server.getStateData(computation, request);
+      } catch (Exception e) {
```

Comment:
I don't think you should catch this, outer block takes care of it

Reply:
Done, but I had to wrap the rethrown exception in a RuntimeException. Otherwise, the compiler complains about declaring checked exceptions. I've tried to declare them, but gave up after 2-3 hops up the call chain.